controllers/secret_controller.go

```diff
@@ -113,7 +113,7 @@ func (s *SecretSyncController) Reconcile(ctx context.Context, req ctrl.Request)
 					}
 					for _, deleteKey := range secretsToDelete {
 						err = remoteClient.Delete(ctx, deleteKey)
-						if err != nil {
+						if err != nil && !errors.IsNotFound(err) {
```

Comment:
How come you removed this `IsNotFound` check?

Reply:
I added it? ;)